cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake

```diff
@@ -1,22 +1,30 @@
 # ToDo: Comment
 function(set_board_compiler_flags COMPILER_FLAGS ARDUINO_DEFINED_VERSION BOARD_ID IS_MANUAL)
 
-    set(COMPILE_FLAGS "-DF_CPU=${${BOARD_ID}.build.f_cpu} -DARDUINO=${ARDUINO_DEFINED_VERSION} -mmcu=${${BOARD_ID}.build.mcu}")
-    if (DEFINED ${BOARD_ID}.build.vid)
-        set(COMPILE_FLAGS "${COMPILE_FLAGS} -DUSB_VID=${${BOARD_ID}.build.vid}")
+    _get_board_property(${BOARD_ID} build.f_cpu FCPU)
+    _get_board_property(${BOARD_ID} build.mcu MCU)
+    set(COMPILE_FLAGS "-DF_CPU=${FCPU} -DARDUINO=${ARDUINO_DEFINED_VERSION} -mmcu=${MCU}")
+
+    _get_board_property_if_exists(${BOARD_ID} build.vid VID)
+    _get_board_property_if_exists(${BOARD_ID} build.pid PID)
+    if (VID)
+        set(COMPILE_FLAGS "${COMPILE_FLAGS} -DUSB_VID=${VID}")
```

Comment:
#4 specifically asks to be able to customize PID and VID on your own. This is not the case here but I think it's beyond the PR's scope. I'll address #4 in a feature branch as soon as the PRs are merged; basically it's just checking whether the user has PID_OVERRIDE/VID_OVERRIDE set and using their values here instead, unless the board has VID and PID predefined, which should yield an error due to the board likely not supporting custom PIDs/VIDs.

Reply:
Oh I got it, thank you for clarifying it for me.

Reply:
I'm not sure what these vid/pid properties work, but in Arduino SDK there are a list of available vid/pid pairs. It could be checked agains of user defined ones, or prints all available vid/pid pairs for the board, if user enters incorrect pid/vid

Reply:
The boards shipping with custom VID/PIDs can be assigned any valid ID, it's just that the predefined ones correspond to registered pairs by Arduino to auto-detect them. For the issue's author, the problem is that he specifically wants nonstandard IDs to easily distinguish multiple arduinos at the same bus using udev.
Warning the user about not using 'proper' IDs is fine, but I think returning an error hardly solves #4.